### DISCUSSION TITLE
Should one use both `proposals` of `polyfill-corejs3` and `shippedProposals` of `@babel/preset-env`?

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#16243] [Q&A] [answered]
As title. Here are the differences:

1. From [docs](https://github.com/babel/babel-polyfills/tree/main/packages/babel-plugin-polyfill-corejs3#proposals) of `proposals`, it says that it will enable all proposals supported by corejs.
2. From [docs](https://babeljs.io/docs/babel-preset-env#shippedproposals) of `shippedProposals` it seems that only those proposals supported by `targets` will be added.

Should these two options be enabled together?

**update**: From their implementation, it seems that `shippedProposals` is a [stricter condition](https://github.com/babel/babel-polyfills/blob/main/packages/babel-plugin-polyfill-corejs3/src/index.ts#L157_L158).

**Comment by JLHwung** [comment, marked as answer]
They are independent options. `shippedProposals` refers to language features supported by Babel, such as decorators (if they were shipped in a browser), while `corejs.proposals` refers to new JavaScript object / methods that you might need a polyfill, such as `Array.prototype.toReversed` (if they were still stage-3 and shipped in a browser).
